src/porepy/numerics/fv/mpsa.py

```diff
@@ -282,14 +282,61 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
                 active_bound, sub_g, l2g_faces
             )
 
+            # Eta can either be a scalar or a vector. If a vector valued eta is passed,
+            # it will have a length equal to the number of subfaces in the entire grid.
+            # If the grid is partitioned into subgrids, we need to adjust the length of
+            # eta to match the subfaces of the subgrid.
+            if isinstance(eta, np.ndarray):
+                # First finding how many subfaces each face is divided into
+                sub_g_subcelltopology = pp.fvutils.SubcellTopology(sub_g)
+                no_unique_subfaces = len(sub_g_subcelltopology.subfno_unique)
+                no_faces = sub_g.cell_faces.shape[0]
+                multiplier = int(no_unique_subfaces / no_faces)
+
+                # Initialize array for the global subface indices
+                global_subgrid_subface_indices = np.zeros(
+                    len(l2g_faces) * multiplier, dtype=int
+                )
+
+                # If all faces are divided into n subfaces, the subfaces will be
```

Comment:
Please avoid duplication of lengthy explanation from mpfa

Reply:
Makes sense, I will remove it